programs/mango-v4/src/health/account_retriever.rs

```diff
@@ -66,19 +68,22 @@ pub fn new_fixed_order_account_retriever<'a, 'info>(
     let expected_ais = active_token_len * 2 // banks + oracles
         + active_perp_len * 2 // PerpMarkets + Oracles
         + active_serum3_len; // open_orders
-    require_msg_typed!(ais.len() == expected_ais, MangoError::InvalidHealthAccountCount,
+    require_msg_typed!(ais.len() >= expected_ais, MangoError::InvalidHealthAccountCount,
         "received {} accounts but expected {} ({} banks, {} bank oracles, {} perp markets, {} perp oracles, {} serum3 oos)",
         ais.len(), expected_ais,
         active_token_len, active_token_len, active_perp_len, active_perp_len, active_serum3_len
     );
+    let fixed_ais = AccountInfoRef::borrow_slice(&ais[..expected_ais])?;
+    let fallback_oracle_ais = AccountInfoRef::borrow_slice(&ais[expected_ais..])?;
 
     Ok(FixedOrderAccountRetriever {
-        ais: AccountInfoRef::borrow_slice(ais)?,
+        ais: fixed_ais,
         n_banks: active_token_len,
         n_perps: active_perp_len,
         begin_perp: active_token_len * 2,
         begin_serum3: active_token_len * 2 + active_perp_len * 2,
         staleness_slot: Some(Clock::get()?.slot),
+        fallback_oracle_ais,
```

Comment:
I don't remember why the FixedOrderAccountRetriever has this awkward form - not needing to allocate a vec and just storing the accountinfos would be preferable - but I'd suggest to keep it consistent.

Meaning: keep a single `ais: Vec<T>` and store `fallback_oracle_start: usize` to know what range of accounts are the fallbacks.

Maybe we can change `ais` type to `&[impl KeyedAccountReader]` or something at some point and avoid the allocation. I think the issue was that if I want to hand out plain references, I need to store the `Ref<'a, &[u8]>` somewhere and the whole thing was frustrating.